test/parallel/test-vfs.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+const common = require('../common');
+
+// This tests the creation of a vfs by monkey-patching fs and Module._stat.
```

Comment:
Shouldn’t we also have a more isolated unit test for `Module._stat` expected behavior, by itself? Outside the context of a full-blown virtual filesystem.

Reply:
@GeoffreyBooth correct, I'm already testing the functionality of pure `Module._stat` before the bfs-specific modification in lines 20 - 30. Does that look good?

Reply:
It's not clear to me where the VFS-specific stuff begins separate from the basic test. Could you maybe move the basic test into its own file?

Reply:
@GeoffreyBooth gotcha, I've added a new file, `test/parallel/test-module-stat.js` that has the isolated test. PTAL.